superbench/benchmarks/micro_benchmarks/cuda_nccl_bw_performance.py

```diff
@@ -0,0 +1,202 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""Module of the NCCL performance benchmarks."""
+
+import os
+import re
+
+from superbench.common.utils import logger
+from superbench.benchmarks import BenchmarkRegistry, Platform, ReturnCode
+from superbench.benchmarks.micro_benchmarks import MicroBenchmarkWithInvoke
+
+
+class CudaNcclBwBenchmark(MicroBenchmarkWithInvoke):
+    """The NCCL bus bandwidth performance benchmark class."""
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+
+        self._bin_name = 'all_reduce_perf'
+        self.__algorithms = {
+            'allreduce': 'all_reduce_perf',
+            'allgather': 'all_gather_perf',
+            'broadcast': 'broadcast_perf',
+            'reduce': 'reduce_perf',
+            'reducescatter': 'reduce_scatter_perf'
+        }
+
+    def add_parser_arguments(self):
+        """Add the specified arguments."""
+        super().add_parser_arguments()
+
+        self._parser.add_argument(
+            '--algo',
+            type=str,
+            nargs='+',
+            default=list(self.__algorithms.keys()),
+            help='Nccl algorithms to benchmark. E.g. {}.'.format(' '.join(list(self.__algorithms.keys()))),
+        )
+        self._parser.add_argument(
+            '--gpu_count',
+            type=int,
+            default=8,
+            help='The count of GPUs used for benchmarking.',
+        )
+        self._parser.add_argument(
+            '--max_size',
+            type=str,
+            default='8192M',
+            help='Max size in bytes to run the nccl test. E.g. 8192M.',
+        )
+        self._parser.add_argument(
+            '--min_size',
+            type=str,
+            default='1',
+            help='Min size in bytes to run the nccl test. E.g. 1.',
+        )
+        self._parser.add_argument(
+            '-f',
```

Comment:
you rename some of nccl-tests parameters while keep others
maybe provide only one `--nccl_tests_args` and set default to `-b 8 -e 8G -f 2 -g 8`?

Reply:
format them to all follow  nccl-tests parameters' name, if only use one argument, it's more easily to failure I think

Reply:
why it's easier to fail? it a wrong parameter is provided, the test will fail in both case.
and could you use long options? I don't think you can pass a `-f` option from config file.

Reply:
besides, do you need to support iters and warmup?